### ISSUE TITLE
argument list for class template "shark::RFClassifier" is missing	

### ISSUE THREAD

**Issue by cydcydcyd** [Shark-ML/Shark#264]
Hi, I download shark4.0 from [https://github.com/Shark-ML/Shark/tree/4.0](url) 
I compile the library and try to use RFTrainer, and I get some errors like:
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)		argument list for class template "shark::RFClassifier" is missing	forestCounting	c:\Users\chenyuduo\Desktop\forestCounting\forestCounting\forestCounting\main.cpp	64	

Severity	Code	Description	Project	File	Line	Suppression State
Error (active)		argument list for class template "shark::RFTrainer" is missing	forestCounting	c:\Users\chenyuduo\Desktop\forestCounting\forestCounting\forestCounting\main.cpp	65	
Can you help me solve it?thanks

**Comment by Ulfgard** [comment]
Have you followed the tutorials?

http://www.shark-ml.org/sphinx_pages/build/html/rest_sources/tutorials/algorithms/rf.html

There is also a random forest example:
https://github.com/Shark-ML/Shark/blob/master/examples/Supervised/RFTutorial.tpp

**Comment by cydcydcyd** [comment]
I have read the tutorials and the example. It doesn't seem to help.
I originally installed version 3.1, but its RFTrainer runs too slowly. I found that someone had the same questions as me. Your suggestion is to update to version 4.0. So I downloaded version 4.0, but encountered the above errors. I found no RFTrainer. cpp in the 4.0 version of src file, but 3.1 does.
I wonder if it matters. And my project is still not compiled successfully.

**Comment by cydcydcyd** [comment]
I have read the tutorials and the example. It doesn't seem to help.
I originally installed version 3.1, but its RFTrainer runs too slowly. I found that someone had the same questions as me. Your suggestion is to update to version 4.0. So I downloaded version 4.0, but encountered the above errors. I found no RFTrainer. cpp in the 4.0 version of src file, but 3.1 does.
I wonder if it matters. And my project is still not compiled successfully.
@Ulfgard

**Comment by Ulfgard** [comment]
I am not seeing your code, so i can only guess. Please note that examples, especially the ones i linkred, required you to define the template arguments:

RFClassifier<unsigned int> model;
RFTrainer<unsigned int> trainer;

**Comment by cydcydcyd** [comment]
@Ulfgard 
Thank you. I read examples to solve the problem.
Another problem is that RFTrainer is still slow in debug mode.
Did you speed up?

**Comment by Ulfgard** [comment]
yeah, try it in release mode and allow your compiler to optimize the code.
________________________________

**Comment by cydcydcyd** [comment]
hello，I try to use RFTrainer as regression, but the result is different from sklearn. 
here is some of my codes:
`	data.shuffle();
	data.makeIndependent();
	RFTrainer<RealVector> trainer(false,false);
	trainer.setMaxDepth(10);
	trainer.setMinSplit(20);
	RFClassifier<RealVector> model;
	trainer.train(model, data);
        Data<RealVector> testInputs = createTestInputs(fMat);
	Data<RealVector> pred = model(testInputs);
`

**Comment by Ulfgard** [comment]
It is different implementations, therefore the result can be different. Also the whole algorithm is stochastic, meaning every run will lead to different results.

How different is it? Are you sure, all parameters are the same?
________________________________

**Comment by cydcydcyd** [comment]
The results are quite different and I'm not sure if I missed the key steps.
I need to use OpenCV to calculate the feature of the images and  the features in  cv::Mat will be converted into shark::RealVector.
`LabeledData<RealVector, RealVector> labeleddata = createLabeledDataFromRange(inputs, outputs);`
`WeightedLabeledData<RealVector, RealVector> data(labeleddata, 1.0f);`
I set the all weights of  features to 1.

**Comment by Ulfgard** [comment]
so what are the results? just for comparison. You don't have to use wieghts, you can just use the labeleddata
________________________________

**Comment by cydcydcyd** [comment]
My input feature is 637 dimensions, and the output tag is 49 dimensions. There are 2000 data in total. It takes 15sec to train with RFTrainer. Is this normal speed? I use release mode.
![image](https://user-images.githubusercontent.com/31031796/53538509-ee1be380-3b48-11e9-9a6f-c6672f479e61.png)
The parameter type of the train function is WeightedLabeledData < RealVector, RealVector >. I don't know how to do without weights.
